sdk/src/resource/resource.cc

```diff
@@ -0,0 +1,66 @@
+#include "opentelemetry/sdk/resource/resource.h"
+#include "opentelemetry/nostd/span.h"
+#include "opentelemetry/sdk/resource/resource_detector.h"
+#include "opentelemetry/version.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace resource
+{
+
+const std::string kTelemetrySdkLanguage = "telemetry.sdk.language";
+const std::string kTelemetrySdkName     = "telemetry.sdk.name";
+const std::string kTelemetrySdkVersion  = "telemetry.sdk.version";
+
+Resource::Resource(const ResourceAttributes &attributes) noexcept : attributes_(attributes) {}
+
+Resource Resource::Merge(const Resource &other) noexcept
+{
+  ResourceAttributes merged_resource_attributes(other.attributes_);
+  for (auto &elem : attributes_)
+  {
+    if (merged_resource_attributes.find(elem.first) == merged_resource_attributes.end())
+    {
+      merged_resource_attributes[elem.first] = elem.second;
+    }
+  }
```

Comment:
I wonder if that could be even more simplified now:
```suggestion
  ResourceAttributes merged_resource_attributes(other.attributes_);
  merged_resource_attributes.insert(attributes_.begin(), attributes_.end());
```